src/components/DeleteAccountModal/DeleteAccountModal.tsx

```diff
@@ -0,0 +1,99 @@
+import { useTranslation } from 'react-i18next';
+import { type SchemaOf, object, string } from 'yup';
+import { useNavigate, useLocation } from 'react-router';
+import { useState } from 'react';
+import { useMutation } from 'react-query';
+
+import PasswordField from '../PasswordField/PasswordField';
+import Button from '../Button/Button';
+
+import styles from './DeleteAccountModal.module.scss';
+
+import type { DeleteAccountFormData } from '#types/account';
+import useForm from '#src/hooks/useForm';
+import { addQueryParam, removeMultipleQueryParams } from '#src/utils/location';
+import { deleteAccountData, logout } from '#src/stores/AccountController';
+
+const DeleteAccountModal = () => {
+  const { t } = useTranslation('user');
+
+  const [enteredPassword, setEnteredPassword] = useState<string>('');
+
+  const deleteAccount = useMutation(deleteAccountData, {
+    onSuccess: async () => {
+      await logout();
+      navigate('/');
+    },
+    onError: () => {
+      setEnteredPassword('');
```

Comment:
When entering a wrong password this modal seems to be closed and reset. Is this clear enough for the user that the entered password was wrong?